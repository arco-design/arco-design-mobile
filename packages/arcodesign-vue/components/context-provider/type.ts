export interface ContextInstallOptions {
    /**
     * 组件名称前缀
     * @en Component prefix
     * @default "Arco"
     */
    componentPrefix?: string;
}

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
}
