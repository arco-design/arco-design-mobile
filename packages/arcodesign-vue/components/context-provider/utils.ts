import { InjectionKey, inject } from 'vue';
import { ContextInstallOptions, GlobalContextParams } from './type';

export const defaultInstallOptions: Required<ContextInstallOptions> = {
    componentPrefix: 'Arco',
};

export const defaultContext: Required<GlobalContextParams> = {
    prefixCls: 'arco',
};

export const configProviderInjectionKey: InjectionKey<GlobalContextParams> =
    Symbol('ArcoContextProvider');

export function getComponentPrefix(options?: ContextInstallOptions) {
    return options?.componentPrefix || defaultInstallOptions.componentPrefix;
}

export function getPrefixCls(componentName?: string) {
    const contextProvider = inject(configProviderInjectionKey, defaultContext);
    const prefix = contextProvider.prefixCls;
    if (componentName) {
        return `${prefix}-${componentName}`;
    }
    return prefix;
}
