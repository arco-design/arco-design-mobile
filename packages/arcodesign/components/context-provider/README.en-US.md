### Others

# ContextProvider 

Global data control component, used to replace global data.

======

> Props

|Property|Description|Type|DefaultValue|
|----------|-------------|------|------|
|prefixCls|Component classname prefix|string|"arco"|
|system|Manually control the current system, and the incoming value will be used directly after being passed in\. It is applicable when the initial value of the system needs to be specified in the ssr scenario\.|"" \| "pc" \| "android" \| "ios"|""|
|useDarkMode|Whether to use dark mode|boolean|false|
|isDarkMode|Whether it is in dark mode|boolean|false|
|theme|Theme variable\. The css variable will be replaced online after input\. The less variable needs to be set|Record\<string, string\>|-|
|locale|Internationalized language configuration|ILocale|-|
|useRtl|Whether to use rtl|boolean|false|

> ILocale

|Property|Description|Type|
|----------|-------------|------|
|locale|Language Type|string|
|LoadMore|-|\{ loadMoreText: string; loadingText: string; prepareText: string; noDataText: string; failLoadText: string; prepareScrollText: string; prepareClickText: string; \}|
|Picker|-|\{ okText: string; cancelText: string; \}|
|Tag|-|\{ addTag: string; \}|
|Dialog|-|\{ okText: string; cancelText: string; \}|
|SwipeLoad|-|\{ normalText: string; activeText: string; \}|
|PullRefresh|-|\{ loadingText: string; pullingText: string; finishText: string; loosingText: string; \}|
|DropdownMenu|-|\{ select: string; \}|
|Pagination|-|\{ previousPage: string; nextPage: string; \}|
|Image|-|\{ loadError: string; \}|
|ImagePicker|-|\{ loadError: string; \}|
|SearchBar|-|\{ placeholder: string; cancelBtn: string; \}|
|Stepper|-|\{ minusButtonName: string; addButtonName: string; \}|
|Keyboard|-|\{ confirm: string; \}|
|Form|-|\{ required: string; type: \{ email: string; url: string; string: string; number: string; array: string; object: string; boolean: string; \}; number: \{ min: string; max: string; equal: string; range: string; positive: string; negative: string; \}; string: \{ \.\.\.; \}; array: \{ \.\.\.; \}; object: \{ \.\.\.; \}; boolean: \{ \.\.\.; \}; \}|
