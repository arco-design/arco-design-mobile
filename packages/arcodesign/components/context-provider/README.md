### 其他

# 全局配置 ContextProvider

全局数据控制组件，用于替换全局数据。

======

> 属性/Props

|参数|描述|类型|默认值|
|----------|-------------|------|------|
|prefixCls|组件类名前缀|string|"arco"|
|system|手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用|"" \| "pc" \| "android" \| "ios"|""|
|useDarkMode|是否使用暗黑模式|boolean|false|
|isDarkMode|是否处于暗黑模式，指定后以指定的值为准|boolean|false|
|theme|主题变量，传入后将在线替换css变量，需设置less变量 @use\-css\-vars: 1|Record\<string, string\>|-|
|locale|国际化语言包配置|ILocale|-|
|useRtl|是否使用Rtl模式|boolean|false|

> ILocale

|参数|描述|类型|
|----------|-------------|------|
|locale|语言类型|string|
|LoadMore|-|\{ loadMoreText: string; loadingText: string; prepareText: string; noDataText: string; failLoadText: string; prepareScrollText: string; prepareClickText: string; \}|
|Picker|-|\{ okText: string; cancelText: string; \}|
|Tag|-|\{ addTag: string; \}|
|Dialog|-|\{ okText: string; cancelText: string; \}|
|SwipeLoad|-|\{ normalText: string; activeText: string; \}|
|PullRefresh|-|\{ loadingText: string; pullingText: string; finishText: string; loosingText: string; \}|
|DropdownMenu|-|\{ select: string; \}|
|Pagination|-|\{ previousPage: string; nextPage: string; \}|
|Image|-|\{ loadError: string; \}|
|ImagePicker|-|\{ loadError: string; \}|
|SearchBar|-|\{ placeholder: string; cancelBtn: string; \}|
|Stepper|-|\{ minusButtonName: string; addButtonName: string; \}|
|Keyboard|-|\{ confirm: string; \}|
|Form|-|\{ required: string; type: \{ email: string; url: string; string: string; number: string; array: string; object: string; boolean: string; \}; number: \{ min: string; max: string; equal: string; range: string; positive: string; negative: string; \}; string: \{ \.\.\.; \}; array: \{ \.\.\.; \}; object: \{ \.\.\.; \}; boolean: \{ \.\.\.; \}; \}|
