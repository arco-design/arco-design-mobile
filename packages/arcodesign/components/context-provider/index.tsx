import React, { createContext, useState, useEffect, useCallback } from 'react';
import { addCssRules, removeCssStyleDom, defaultLocale, ILocale } from '@arco-design/mobile-utils';

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
    /**
     * 手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用
     * @en Manually control the current system, and the incoming value will be used directly after being passed in. It is applicable when the initial value of the system needs to be specified in the ssr scenario.
     * @default ""
     */
    system?: 'pc' | 'android' | 'ios' | '';
    /**
     * 是否使用暗黑模式
     * @en Whether to use dark mode
     * @default false
     */
    useDarkMode?: boolean;
    /**
     * 是否处于暗黑模式，指定后以指定的值为准
     * @en Whether it is in dark mode
     * @default false
     */
    isDarkMode?: boolean;
    /**
     * 主题变量，传入后将在线替换css变量，需设置less变量 @use-css-vars: 1
     * @en Theme variable. The css variable will be replaced online after input. The less variable needs to be set @use-css-vars: 1
     */
    theme?: Record<string, string>;
    /**
     * 国际化语言包配置
     * @en Internationalized language configuration
     */
    locale?: ILocale;
    /**
     * 是否使用Rtl模式
     * @en Whether to use rtl
     * @default false
     */
    useRtl?: boolean;
}

export const defaultContext: GlobalContextParams = {
    prefixCls: 'arco',
    system: '',
    useDarkMode: false,
    isDarkMode: false,
    locale: defaultLocale,
    useRtl: false,
};

export const GlobalContext = createContext<GlobalContextParams>(defaultContext);

export interface ContextProviderProps extends GlobalContextParams {
    children: React.ReactNode;
}

export type WithGlobalContext<T> = T & { context?: GlobalContextParams };

/**
 * 全局数据控制组件，用于替换全局数据。
 * @en Global data control component, used to replace global data.
 * @type 其他
 * @type_en Others
 * @name 全局配置
 * @name_en ContextProvider
 */
export default function ContextProvider(props: ContextProviderProps) {
    const {
        children,
        useDarkMode,
        isDarkMode,
        theme,
        locale = defaultLocale,
        ...restProps
    } = props;
    const [isDarkModeState, setIsDarkModeState] = useState<boolean>();
    const changeDarkMode = useCallback(
        (res: MediaQueryListEvent) => setIsDarkModeState(res.matches),
        [],
    );

    useEffect(() => {
        const matchMedia = window.matchMedia('(prefers-color-scheme: dark)');
        if (useDarkMode) {
            const dark = matchMedia.matches;
            setIsDarkModeState(dark);
            if (typeof matchMedia.addEventListener === 'function') {
                matchMedia.addEventListener('change', changeDarkMode);
            } else if (typeof matchMedia.addListener === 'function') {
                matchMedia.addListener(changeDarkMode);
            }
        }
        return () => {
            if (useDarkMode) {
                if (typeof matchMedia.removeEventListener === 'function') {
                    matchMedia.removeEventListener('change', changeDarkMode);
                } else if (typeof matchMedia.removeListener === 'function') {
                    matchMedia.removeListener(changeDarkMode);
                }
            }
        };
    }, [useDarkMode]);

    useEffect(() => {
        if (!theme) {
            return;
        }
        addCssRules('arcoCustomTheme', theme);
        return () => {
            removeCssStyleDom('arcoCustomTheme');
        };
    }, [theme]);

    return (
        <GlobalContext.Provider
            value={{
                ...defaultContext,
                ...restProps,
                isDarkMode: isDarkMode === void 0 ? isDarkModeState : isDarkMode,
                useDarkMode,
                locale: locale || defaultLocale,
            }}
        >
            {children}
        </GlobalContext.Provider>
    );
}

export const ContextLayout = GlobalContext.Consumer;

export function CompWithGlobalContext<P extends JSX.IntrinsicAttributes>(
    Component: React.FunctionComponent<P>,
) {
    return function (props: WithGlobalContext<P>) {
        const { context: propsContext, ...others } = props;
        return (
            <ContextLayout>
                {context => (
                    <ContextProvider {...{ ...context, ...propsContext }}>
                        <Component {...(others as P)} />
                    </ContextProvider>
                )}
            </ContextLayout>
        );
    };
}
