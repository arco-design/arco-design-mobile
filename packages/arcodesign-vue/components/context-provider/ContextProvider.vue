<script lang="ts" setup>
import { provide } from 'vue';
import { GlobalContextParams } from './type';
import { configProviderInjectionKey, defaultContext } from './utils';

const props = withDefaults(defineProps<GlobalContextParams>(), defaultContext);
provide(configProviderInjectionKey, props);
</script>
<template>
    <slot />
</template>
