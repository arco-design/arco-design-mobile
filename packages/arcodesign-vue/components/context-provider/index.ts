import ContextProvider from './ContextProvider.vue';
import { installFactory } from '../_helpers/install';

export type * from './type';
export * from './utils';

const install = installFactory(ContextProvider);

export { ContextProvider };

export default {
    install,
};
