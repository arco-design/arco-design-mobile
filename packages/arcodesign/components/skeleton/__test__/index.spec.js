import React from 'react';
import { mount } from 'enzyme';
import demoTest from '../../../tests/demoTest';
import mountTest from '../../../tests/mountTest';
import { defaultContext } from '../../context-provider';
import Skeleton from '..';

demoTest('skeleton');

mountTest(Skeleton, 'Skeleton');

const prefix = `${defaultContext.prefixCls}-skeleton`;

describe('Skeleton', () => {
    it('should render correctly when set ref', () => {
        const ref = React.createRef();
        const titleRef = React.createRef();
        const paragraphRef = React.createRef();
        const avatarRef = React.createRef();
        const nodeRef = React.createRef();
        mount(
            <Skeleton ref={ref}>
                <Skeleton.Title ref={titleRef} />
                <Skeleton.Paragraph ref={paragraphRef} />
                <Skeleton.Avatar ref={avatarRef} />
                <Skeleton.Node ref={nodeRef}>
                    <div />
                </Skeleton.Node>
            </Skeleton>,
        );
        expect(ref.current.dom).not.toBeUndefined();
        expect(titleRef.current.dom).not.toBeUndefined();
        expect(paragraphRef.current.dom).not.toBeUndefined();
        expect(avatarRef.current.dom).not.toBeUndefined();
        expect(nodeRef.current.dom).not.toBeUndefined();
    });
    it('should render correctly when set avatar', () => {
        const comp = mount(<Skeleton avatar />);
        expect(comp.find(`.${prefix}-avatar`).length).toBe(1);
    });
    it('should render correctly when set grid', () => {
        const comp = mount(<Skeleton grid={{ columns: 5 }} />);
        expect(comp.find(`.${prefix}-grid-item`).length).toBe(5);
    });
    it('should render correctly when set children', () => {
        const comp = mount(
            <Skeleton>
                <Skeleton.Node>
                    <div style={{ width: 20, height: 20 }} />
                </Skeleton.Node>
            </Skeleton>,
        );
        expect(comp.find(`.${prefix}-node`).length).toBe(1);
        expect(comp.find(`.${prefix}-title`).length).toBe(1);
        expect(comp.find(`.${prefix}-paragraph`).length).toBe(1);
    });
    it('should render correctly when set animation', () => {
        const comp = mount(
            <Skeleton>
                <Skeleton.Node>
                    <div style={{ width: 20, height: 20 }} />
                </Skeleton.Node>
            </Skeleton>,
        );
        expect(comp.find(`.${prefix}-animation-gradient`).length).toBe(5);
        expect(comp.find(`.${prefix}-animation-breath`).length).toBe(0);
        comp.setProps({
            animation: 'breath',
        });
        expect(comp.find(`.${prefix}-animation-gradient`).length).toBe(0);
        expect(comp.find(`.${prefix}-animation-breath`).length).toBe(5);
        comp.setProps({
            showAnimation: false,
        });
        expect(comp.find(`.${prefix}-animation-gradient`).length).toBe(0);
        expect(comp.find(`.${prefix}-animation-breath`).length).toBe(0);
    });
});
