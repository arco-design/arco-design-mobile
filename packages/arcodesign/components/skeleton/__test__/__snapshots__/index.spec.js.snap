// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`skeleton demo test skeleton demo: animation.md renders correctly 1`] = `
<div>
  <div
    class="arco-cell-group all-border-box"
  >
    <div
      class="cell-group-body bordered"
    >
      <div
        class="arco-cell all-border-box bordered"
      >
        <div
          class="arco-cell-inner"
        >
          <div
            class="cell-label"
          >
            <div
              class="cell-title"
            >
              show animation
            </div>
          </div>
          <div
            class="cell-content has-label"
          >
            <div
              aria-checked="true"
              aria-disabled="false"
              class="arco-switch all-border-box type-pc system-pc fully checked"
              role="switch"
              tabindex="0"
            >
              <div
                class="arco-switch-inner"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="arco-cell all-border-box bordered"
      >
        <div
          class="arco-cell-inner"
        >
          <div
            class="cell-label"
          >
            <div
              class="cell-title"
            >
              type
            </div>
          </div>
          <div
            class="cell-content has-label"
          >
            <div
              class="arco-radio-group"
            >
              <div
                aria-checked="true"
                aria-disabled="false"
                class="arco-radio shape-circle inline"
                role="checkbox"
                tabindex="0"
              >
                <div
                  class="radio-icon checked"
                >
                  <svg
                    class="arco-icon arco-icon-circle-checked "
                    height="1em"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M10 1c-5 0-9 4-9 9s4 9 9 9 9-4 9-9-4-9-9-9zm4.9 6.3L10 13.2s0 .1-.1.1l-.6.7c-.1.1-.2.1-.3.2-.1 0-.3 0-.4-.1l-.6-.7-.1-.1-2.8-2.4c-.2-.2-.2-.5-.1-.7l.6-.7c.2-.2.5-.2.7-.1l2.5 2.1L13.5 6c.2-.2.5-.2.7-.1l.7.6c.1.3.2.6 0 .8z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
                <div
                  class="radio-text"
                >
                  gradient
                </div>
              </div>
              <div
                aria-checked="false"
                aria-disabled="false"
                class="arco-radio shape-circle inline"
                role="checkbox"
                tabindex="0"
              >
                <div
                  class="radio-icon"
                >
                  <svg
                    class="arco-icon arco-icon-circle-unchecked "
                    height="1em"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10 19c-5 0-9-4-9-9s4-9 9-9 9 4 9 9-4 9-9 9zm0-17c-4.4 0-8 3.6-8 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <div
                  class="radio-text"
                >
                  breath
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="arco-skeleton"
    style="color:rgba(148, 191, 255, 20%)"
  >
    <div
      class="arco-skeleton-content"
    >
      <div
        class="arco-skeleton-item arco-skeleton-title arco-skeleton-animation-gradient"
        style="width:40%;background-color:"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
      <div
        class="arco-skeleton-paragraph"
      >
        <div
          class="arco-skeleton-item arco-skeleton-paragraph-line arco-skeleton-animation-gradient"
          style="background-color:"
        >
          <div
            class="arco-skeleton-animation-item"
            style="left:0"
          />
        </div>
        <div
          class="arco-skeleton-item arco-skeleton-paragraph-line arco-skeleton-animation-gradient"
          style="background-color:"
        >
          <div
            class="arco-skeleton-animation-item"
            style="left:0"
          />
        </div>
        <div
          class="arco-skeleton-item arco-skeleton-paragraph-line arco-skeleton-animation-gradient"
          style="width:60%;background-color:"
        >
          <div
            class="arco-skeleton-animation-item"
            style="left:0"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`skeleton demo test skeleton demo: avatar.md renders correctly 1`] = `
<div
  class="arco-skeleton arco-skeleton-with-avatar"
>
  <div
    class="arco-skeleton-item arco-skeleton-avatar arco-skeleton-animation-gradient"
  >
    <div
      class="arco-skeleton-animation-item"
      style="left:0"
    />
  </div>
  <div
    class="arco-skeleton-content"
  >
    <div
      class="arco-skeleton-item arco-skeleton-title arco-skeleton-animation-gradient"
      style="width:40%"
    >
      <div
        class="arco-skeleton-animation-item"
        style="left:0"
      />
    </div>
    <div
      class="arco-skeleton-paragraph"
    >
      <div
        class="arco-skeleton-item arco-skeleton-paragraph-line arco-skeleton-animation-gradient"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
      <div
        class="arco-skeleton-item arco-skeleton-paragraph-line arco-skeleton-animation-gradient"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
      <div
        class="arco-skeleton-item arco-skeleton-paragraph-line arco-skeleton-animation-gradient"
        style="width:60%"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`skeleton demo test skeleton demo: custom.md renders correctly 1`] = `
<div>
  <div
    class="arco-cell-group all-border-box"
  >
    <div
      class="cell-group-body bordered"
    >
      <div
        class="arco-cell all-border-box bordered"
      >
        <div
          class="arco-cell-inner"
        >
          <div
            class="cell-label"
          >
            <div
              class="cell-title"
            >
              loading
            </div>
          </div>
          <div
            class="cell-content has-label"
          >
            <div
              aria-checked="true"
              aria-disabled="false"
              class="arco-switch all-border-box type-pc system-pc fully checked"
              role="switch"
              tabindex="0"
            >
              <div
                class="arco-switch-inner"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="arco-skeleton"
  >
    <div
      class="arco-skeleton-grid"
      style="margin-bottom:20px"
    >
      <div
        class="arco-skeleton-grid-item"
      >
        <div
          class="arco-skeleton-item arco-skeleton-grid-icon arco-skeleton-animation-gradient"
        >
          <div
            class="arco-skeleton-animation-item"
            style="left:0"
          />
        </div>
        <div
          class="arco-skeleton-item arco-skeleton-grid-text arco-skeleton-animation-gradient"
        >
          <div
            class="arco-skeleton-animation-item"
            style="left:0"
          />
        </div>
      </div>
      <div
        class="arco-skeleton-grid-item"
      >
        <div
          class="arco-skeleton-item arco-skeleton-grid-icon arco-skeleton-animation-gradient"
        >
          <div
            class="arco-skeleton-animation-item"
            style="left:0"
          />
        </div>
        <div
          class="arco-skeleton-item arco-skeleton-grid-text arco-skeleton-animation-gradient"
        >
          <div
            class="arco-skeleton-animation-item"
            style="left:0"
          />
        </div>
      </div>
      <div
        class="arco-skeleton-grid-item"
      >
        <div
          class="arco-skeleton-item arco-skeleton-grid-icon arco-skeleton-animation-gradient"
        >
          <div
            class="arco-skeleton-animation-item"
            style="left:0"
          />
        </div>
        <div
          class="arco-skeleton-item arco-skeleton-grid-text arco-skeleton-animation-gradient"
        >
          <div
            class="arco-skeleton-animation-item"
            style="left:0"
          />
        </div>
      </div>
    </div>
    <div
      style="display:flex;justify-content:space-between;align-items:flex-start;margin-bottom:20px"
    >
      <div
        class="arco-skeleton-item arco-skeleton-avatar arco-skeleton-animation-gradient"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
      <div
        class="arco-skeleton-item arco-skeleton-node arco-skeleton-animation-gradient"
        style="border-radius:10px"
      >
        <div
          style="width:250px;height:50px"
        />
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
    </div>
    <div
      class="arco-skeleton-item arco-skeleton-title arco-skeleton-animation-gradient"
      style="width:40%;margin-bottom:20px"
    >
      <div
        class="arco-skeleton-animation-item"
        style="left:0"
      />
    </div>
    <div
      class="arco-skeleton-paragraph"
      style="margin-bottom:20px"
    >
      <div
        class="arco-skeleton-item arco-skeleton-paragraph-line arco-skeleton-animation-gradient"
        style="width:100%"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
      <div
        class="arco-skeleton-item arco-skeleton-paragraph-line arco-skeleton-animation-gradient"
        style="width:100%"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
      <div
        class="arco-skeleton-item arco-skeleton-paragraph-line arco-skeleton-animation-gradient"
        style="width:100%"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
      <div
        class="arco-skeleton-item arco-skeleton-paragraph-line arco-skeleton-animation-gradient"
        style="width:80%"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
      <div
        class="arco-skeleton-item arco-skeleton-paragraph-line arco-skeleton-animation-gradient"
        style="width:60%"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`skeleton demo test skeleton demo: grid.md renders correctly 1`] = `
<div
  class="arco-skeleton"
>
  <div
    class="arco-skeleton-grid"
  >
    <div
      class="arco-skeleton-grid-item"
    >
      <div
        class="arco-skeleton-item arco-skeleton-grid-icon arco-skeleton-animation-gradient"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
      <div
        class="arco-skeleton-item arco-skeleton-grid-text arco-skeleton-animation-gradient"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
    </div>
    <div
      class="arco-skeleton-grid-item"
    >
      <div
        class="arco-skeleton-item arco-skeleton-grid-icon arco-skeleton-animation-gradient"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
      <div
        class="arco-skeleton-item arco-skeleton-grid-text arco-skeleton-animation-gradient"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
    </div>
    <div
      class="arco-skeleton-grid-item"
    >
      <div
        class="arco-skeleton-item arco-skeleton-grid-icon arco-skeleton-animation-gradient"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
      <div
        class="arco-skeleton-item arco-skeleton-grid-text arco-skeleton-animation-gradient"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
    </div>
    <div
      class="arco-skeleton-grid-item"
    >
      <div
        class="arco-skeleton-item arco-skeleton-grid-icon arco-skeleton-animation-gradient"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
      <div
        class="arco-skeleton-item arco-skeleton-grid-text arco-skeleton-animation-gradient"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`skeleton demo test skeleton demo: index.md renders correctly 1`] = `
<div
  class="arco-skeleton"
>
  <div
    class="arco-skeleton-content"
  >
    <div
      class="arco-skeleton-item arco-skeleton-title arco-skeleton-animation-gradient"
      style="width:40%"
    >
      <div
        class="arco-skeleton-animation-item"
        style="left:0"
      />
    </div>
    <div
      class="arco-skeleton-paragraph"
    >
      <div
        class="arco-skeleton-item arco-skeleton-paragraph-line arco-skeleton-animation-gradient"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
      <div
        class="arco-skeleton-item arco-skeleton-paragraph-line arco-skeleton-animation-gradient"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
      <div
        class="arco-skeleton-item arco-skeleton-paragraph-line arco-skeleton-animation-gradient"
        style="width:60%"
      >
        <div
          class="arco-skeleton-animation-item"
          style="left:0"
        />
      </div>
    </div>
  </div>
</div>
`;
